google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/project/ProjectSelectionDialog.java

```diff
@@ -359,11 +359,14 @@ private void hideSignInRequest() {
   }
 
   private void validateProjectSelection() {
-    if (projectListTable.getSelectedRow() >= 0) {
+    int selectedRow = projectListTable.getSelectedRow();
+    // make explicit visible row check due to non-standard behaivor of IDEA's TableSpeedSearch
+    int visibleRowCount = projectListTable.getRowSorter().getViewRowCount();
+    if (selectedRow >= 0 && selectedRow < visibleRowCount) {
       dialogWrapper.setOKActionEnabled(true);
       selectedProjectsByAccount.put(
           (CredentialedUser) accountComboBox.getSelectedItem(), getSelectedProjectName());
-    } else {
```

Comment:
just so I understand; if none of these conditions are met, this just represents a non-existent index and nothing should be done? We don't want to disable the ok action in this case?

Reply:
It could be an existing index but generated by IDEA table speed search and out of bounds for normal table filtering. So we check explicitly for no selection (<0) to avoid disabling ok action when some selection is present.